bowling-ball/Game.cs

```diff
@@ -1,17 +1,141 @@
+                //  last value is the extra roll and the value before that is the last frame roll chance (=10)
+                else if ( pins[lengthOfPinsArray - 3] == KnockedDownTen)
+                {
+                    Frames[PointerToFrame].SetExtraRollValue(pins[lengthOfPinsArray - 1]);
+                    Frames[PointerToFrame].SetFrame(0, pins[lengthOfPinsArray - 3]);
+                    Frames[PointerToFrame].SetFrame(1, pins[lengthOfPinsArray - 2]);
+                }
+                else
+                {
+                    Frames[PointerToFrame].SetFrame(0, pins[lengthOfPinsArray - 2]);
+                    Frames[PointerToFrame].SetFrame(1, pins[lengthOfPinsArray - 1]);
+                }
+
+            }
+        }
+
+        public int GetScore()
+        {
+            int totalScore = 0;
+            const int SecondLastFrame = 8;
+
+            for(int CurrentPointer=0; CurrentPointer < PointerToFrame; CurrentPointer++)
+            {
+                // Checking for strike
+                if (Frames[CurrentPointer].IsStrike() == true && Frames[CurrentPointer].CheckForEmptyFrame() == false)
+                {
+                    // If it is not SecondLastFrame and next frame is strike again, then adding two strikes and first roll of CurrentPointer + 2
+                    if (CurrentPointer != SecondLastFrame && Frames[CurrentPointer + 1].GetFrame(0) == KnockedDownTen)
+                            totalScore += KnockedDownTen + KnockedDownTen + Frames[CurrentPointer + 2].GetFrame(0);
+                    // If it is second last frame and last frame is strike, then adding two strikes and immediate next roll
+                    else if(CurrentPointer == SecondLastFrame && Frames[CurrentPointer + 1].GetFrame(0) == KnockedDownTen)
+                            totalScore += KnockedDownTen + KnockedDownTen + Frames[CurrentPointer + 1].GetFrame(1);
+                    // If neither of the above conditions. then adding strike and next frames bonus
+                    else
+                        totalScore += KnockedDownTen + Frames[CurrentPointer + 1].StrikeBonus(false);
+                }
+                // Checking for spare
+                else if (Frames[CurrentPointer].IsSpare() == true && Frames[CurrentPointer].CheckForEmptyFrame() == false)
+                    totalScore += KnockedDownTen + Frames[CurrentPointer + 1].SpareBonus(false);
+                // Neither strike nor spare
+                else if (Frames[CurrentPointer].IsSpare() == false && Frames[CurrentPointer].IsStrike() == false 
+                    && Frames[CurrentPointer].CheckForEmptyFrame() == false)
+                    totalScore += Frames[CurrentPointer].CalculateScore();
+                
+            }
+
+            //Checking whether LastFrame has strike or spare and adding extra roll value to totalScores 
+            if (PointerToFrame == TotalPossibleFrames - 1)

```

Comment:
`Frames[PointerToFrame].CheckForEmptyFrame() == false` could have been checked once at this place.